### PR TITLE
[QUAD-196]: Added the ability to view all the public networks in quadriga.

### PR DESCRIPTION
**Pull request by suprajkolluri**
Added the ability to view all the public networks in quadriga in a single page in quadriga/sites section.

### CODE REVIEW

File: Quadriga/src/main/java/edu/asu/spring/quadriga/dao/impl/NetworkDAO.java [hunk 1]
```diff
@@ -294,6 +294,24 @@ public String addNetworkStatement(String rowid, String networkId, String id, Str
...
+    public List<NetworksDTO> getApprovedNetworkList() throws QuadrigaStorageException {
+        try {
+            Query query = sessionFactory.getCurrentSession().getNamedQuery("NetworksDTO.findApprovedNetworks");
+
+            @SuppressWarnings("unchecked")
+            List<NetworksDTO> listNetworksDTO = query.list();
+
+            return listNetworksDTO;
+        } catch (Exception e) {
+            logger.error("Error in fetching a network status: ", e);
```
Comment (jdamerow) [inline Quadriga/src/main/java/edu/asu/spring/quadriga/dao/impl/NetworkDAO.java:311]: remove logger. exception will be logged when handled. And does the `list` method really throw just `Exception`?

File: Quadriga/src/main/java/edu/asu/spring/quadriga/rest/open/NodeInfoController.java [hunk 2]
```diff
@@ -26,72 +26,79 @@
...
-                    if (occur.getContents().length() < end + offset) {
-                        end = occur.getContents().length() -1;
-                    } else {
-                        end = end + offset;
-                    }
-                    
-                    String cutContent = occur.getContents().substring(start, end);
-                    occurance.append("textContent", cutContent);
-                    
+            if (occur.getProject().getUnixName().equals(projectUnix.trim()) || "".equals(projectUnix.trim())) {
```
Comment (jdamerow) [inline Quadriga/src/main/java/edu/asu/spring/quadriga/rest/open/NodeInfoController.java:44]: why not use the `isEmpty` method?

File: Quadriga/src/main/java/edu/asu/spring/quadriga/rest/open/NodeInfoController.java [hunk 3]
```diff
@@ -26,72 +26,79 @@
...
+                        && occur.getProject().getProjectAccess() != EProjectAccessibility.PUBLIC) {
+                    flag = false;
                 }
-                for (TextPhrase phrase : phrases) {
-                    if (phrase.getFormattedPointer() == null || phrase.getFormattedPointer().trim().isEmpty()) {
-                        JSONObject jsonPhrase = new JSONObject();
-                        int position = phrase.getPosition();
-                        if (position > start) {
-                            position -= start;
+                if (flag) {
```
Comment (jdamerow) [inline Quadriga/src/main/java/edu/asu/spring/quadriga/rest/open/NodeInfoController.java:50]: why do you set a flag? Why not do the condition check right here? Also, that only text occurrences of public projects should be returned, is always true. So you can always check for that, not just when there is no unix name.

File: Quadriga/src/main/webapp/WEB-INF/tiles/layouts/skeleton_public_sites.jsp [hunk 4]
```diff
@@ -108,18 +111,34 @@
...
-        <nav>
-          <ul class="nav pull-right">
-            <tiles:insertAttribute name="navigation" />
-          </ul>
-        </nav>
+        <div class="navbar-collapse collapse navbar-right">
+					<ul class="nav navbar-nav">
+						<tiles:insertAttribute name="navigation" />
+					</ul>
+		</div>
```
Comment (jdamerow) [inline Quadriga/src/main/webapp/WEB-INF/tiles/layouts/skeleton_public_sites.jsp:118]: please fix indentation

File: Quadriga/src/main/webapp/WEB-INF/tiles/layouts/skeleton_public_sites.jsp [hunk 5]
```diff
@@ -108,18 +111,34 @@
...
+			</div>
+			</div>
+			<div  class="col-md-3" style="text-align:center">
+					<!-- Sidebar -->
+					<!-- Recent Posts -->
+					<tiles:insertAttribute name="right-navigation" />
+					<!-- /Recent Posts -->
+					<!-- /Sidebar -->
+			</div>
+			</div>
```
Comment (jdamerow) [inline Quadriga/src/main/webapp/WEB-INF/tiles/layouts/skeleton_public_sites.jsp:141]: here too

File: Quadriga/src/main/webapp/WEB-INF/views/public/sites/nav-networkrightplane.jsp [hunk 6]
```diff
@@ -0,0 +1,84 @@
...
+
+	<div class="text-left">
+		<h5>
+			<a href="#" id="exportPng"><i class="fa fa-file-image-o"
+				aria-hidden="true" title="Download as PNG"></i> Download network as
+				PNG</a><br> <small>Image will open in a new tab or window.</small>
+		</h5>
+	</div>
+
+</c:if>
```
Comment (jdamerow) [inline Quadriga/src/main/webapp/WEB-INF/views/public/sites/nav-networkrightplane.jsp:84]: don't indent html that much please

**Comment by jdamerow** [comment]
ok to test

**Comment by jdamerow** [comment]
There should be no whitespace on the right on this page:
image

**Comment by jdamerow** [comment]
Menu entries:
- Home should be "All Projects"
- Public networks should be "Explore the Graph"

**Comment by jdamerow** [comment]
And since you're on it, would you please add a little loading icon in the concept description box and text occurrences box while the content for those is being retrieved? Otherwise, looks great!

**Review by jdamerow — CHANGES_REQUESTED** [review]

File: Quadriga/src/main/java/edu/asu/spring/quadriga/dao/impl/NetworkDAO.java (hunk 1 above)
Comment (suprajkolluri) [reply Quadriga/src/main/java/edu/asu/spring/quadriga/dao/impl/NetworkDAO.java:311]: It throws HibernateException, do you want to me to wrap it in QuadrigaStorage exception leave it for the calling method to handle?

File: Quadriga/src/main/webapp/WEB-INF/tiles/layouts/skeleton_public_sites.jsp (hunk 5 above)
Comment (suprajkolluri) [inline Quadriga/src/main/webapp/WEB-INF/tiles/layouts/skeleton_public_sites.jsp]: I am using the IDE XML formatter, do we have any format template for XMLs?

File: Quadriga/src/main/java/edu/asu/spring/quadriga/dao/impl/NetworkDAO.java (hunk 1 above)
Comment (jdamerow) [reply Quadriga/src/main/java/edu/asu/spring/quadriga/dao/impl/NetworkDAO.java:311]: yes, and always catch specific exceptions, never just exception.

File: Quadriga/src/main/webapp/WEB-INF/views/public/sites/nav-networkrightplane.jsp (hunk 6 above)
Comment (suprajkolluri) [reply Quadriga/src/main/webapp/WEB-INF/views/public/sites/nav-networkrightplane.jsp:84]: For indentation I am using the IDE XML Formatting, do we have a template/specification for XMLs too?

**Pull request closed without merging**